astropy/io/fits/src/compressionmodule.c

```diff
@@ -1016,6 +1016,96 @@ PyObject* compression_compress_hdu(PyObject* self, PyObject* args)
 }
 
 
+// This is a cleaned-up version of the cfitio 3.41 function
+// 'fits_get_img_equivtype' (AKA 'ffgiet') with a slightly different interface.
+// It's necessary to copy this code because the original code reads the header,
+// but our munged fitsfile doesn't have any header to read.
+static void get_equiv_datatype(
+    int bitpix,  // BITPIX for image
+    double bscale,  // BSCALE for image
+    double bzero,  // BZERO for image
+    int *imgtype,  // cfitsio image type to use
+    int *numpytype  // numpy image type to use
+    )
+{
+    double min_val;
+    double max_val;
+    long lngzero;
+    long lngscale;
+
+    bitpix_to_datatypes(bitpix, imgtype, numpytype);
+
+    if (bscale == 1.0 && bzero == 0.0) {
+        return;  /* no scaling */
+    }
+
+    switch (*imgtype)
+    {
+      case BYTE_IMG:   /* 8-bit image */
+        min_val = 0.;
+        max_val = 255.0;
+        break;
+      case SHORT_IMG:
+        min_val = -32768.0;
+        max_val =  32767.0;
+        break;
+      case LONG_IMG:
+        min_val = -2147483648.0;
+        max_val =  2147483647.0;
+        break;
+      default:  /* don't have to deal with other data types */
```

Comment:
The `bitpix_to_datatypes` could also return `TLONGLONG`. That could warrant it's own case.

Reply:
cfitsio won't compress 64-bit integers. I'll put in a trap for it though.